### PR TITLE
DROOLS-7272 DSL: Extract Drools from Kogito pipelines

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/DROOLS-7272

Related:
- https://github.com/kiegroup/drools/pull/4945
- https://github.com/kiegroup/kogito-pipelines/pull/790
- https://github.com/kiegroup/drools-website/pull/239

### CODE REVIEW

**Review by mbiarnes — APPROVED** [review]

File: .ci/jenkins/dsl/test.sh [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+elif [[ "${git_url}" = git@* ]]; then 
+  git_server_url="git@${GIT_SERVER}:"
+else
+  echo "Unknown protocol for url ${git_url}"
+  exit 1
+fi
+
+git_author="$(echo ${git_url} | awk -F"${git_server_url}" '{print $2}' | awk -F. '{print $1}'  | awk -F/ '{print $1}')"
+
+export DSL_DEFAULT_MAIN_CONFIG_FILE_REPO="${git_author}"/optaplanner
```
Comment (lampajr) [inline .ci/jenkins/dsl/test.sh:25]: Shouldn't be pointing to drools?

```suggestion
export DSL_DEFAULT_MAIN_CONFIG_FILE_REPO="${git_author}"/drools
```

If yes, also the following lines should be updated too.

File: .ci/jenkins/dsl/test.sh [hunk 2]
```diff
@@ -1,4 +1,32 @@
...
+export DSL_DEFAULT_FALLBACK_MAIN_CONFIG_FILE_REPO=kiegroup/optaplanner
```
Comment (lampajr) [inline .ci/jenkins/dsl/test.sh:26]:
```suggestion
export DSL_DEFAULT_FALLBACK_MAIN_CONFIG_FILE_REPO=kiegroup/drools
```

File: .ci/jenkins/dsl/test.sh [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+export DSL_DEFAULT_MAIN_CONFIG_FILE_PATH=.ci/jenkins/config/main.yaml
+export DSL_DEFAULT_BRANCH_CONFIG_FILE_REPO="${git_author}"/optaplanner
```
Comment (lampajr) [inline .ci/jenkins/dsl/test.sh:28]:
```suggestion
export DSL_DEFAULT_BRANCH_CONFIG_FILE_REPO="${git_author}"/drools
```

**Review by lampajr — CHANGES_REQUESTED** [review]

**Review by mbiarnes — APPROVED** [review]
LGTM

**Review by lampajr — APPROVED** [review]

**Pull request merged**